RHIDP-5959 Update dangerouslyAllowSignInWithoutUserInCatalog config (#978)
redhat-developer/red-hat-developers-documentation-rhdh: modules/authentication/proc-enabling-authentication-with-microsoft-azure.adoc

`AUTH_AZURE_CLIENT_SECRET`:: Enter your saved *Application (client) secret*.

. Set up the Microsoft Azure authentication provider in your {product-short} custom configuration, such as `app-config-rhdh`:
.. Configure mandatory fields:
+
.`app-config-rhdh.yaml` fragment
[source,yaml,subs="+quotes,+attributes"]
----
auth:
  environment: production # <1>
  providers:
    microsoft:
      production:
        clientId: ${AUTH_AZURE_CLIENT_ID} # <2>
        clientSecret: ${AUTH_AZURE_CLIENT_SECRET}
        tenantId: ${AUTH_AZURE_TENANT_ID}
signInPage: microsoft # <3>
----

> <1> Mark the environment as production to hide the **Guest** login in the {product-short} home page.

Author: ```suggestion
<1> Mark the environment as production to hide the **Guest** login in the {product-short} login page.
```

> <1> Mark the environment as production to hide the **Guest** login in the {product-short} login page.

Reviewer: ```suggestion
<1> Mark the environment as production and disable the **Guest** login option in the {product-short} login page.
```

> <2> Apply the same Microsoft Azure credentials configured in your {product-short} secrets.

Reviewer 2: ```suggestion
<2> Apply the Microsoft Azure credentials configured in your {product-short} secrets.
```

> <2> Use the {product-short} application information that you have created in Microsoft Azure and configured in OpenShift as secrets.

Reviewer: ```suggestion
<2> Apply the same Microsoft Azure credentials configured in your {product-short} secrets.
```

> <3> Enable the Microsoft Azure provider as default sign-in provider.

Reviewer: ```suggestion
<3> Set the Microsoft Azure provider as your {product-short} sign-in provider.
```

        sessionDuration: { hours: 24 }
----

`signIn` ::

`resolvers`:::

> Enter the resolver list to override the default resolver: `emailLocalPartMatchingUserEntityName`. The authentication provider tries each sign-in resolver until it succeeds, and fails if none succeed.

Author: ```suggestion
After successful authentication, the user signing in must be resolved to an existing user in the {product-short} catalog. To best match users securely for your use case, consider configuring a specific resolver. Enter the resolver list to override the default resolver: `emailLocalPartMatchingUserEntityName`.
+
The authentication provider tries each sign-in resolver in order until it succeeds, and fails if none succeed.
+
WARNING: In production mode, only configure one resolver to ensure users are securely matched. 
```

`resolver`::::
Enter the sign-in resolver name.
Available values:
* `userIdMatchingUserEntityAnnotation`
* `emailLocalPartMatchingUserEntityName`
* `emailMatchingUserEntityProfileEmail`

`dangerouslyAllowSignInWithoutUserInCatalog: true`::::
+

> Configure the sign-in resolver to enable authentication without requiring to provision users in the {product-short} software catalog.

Author: As suggested above:

```suggestion
Configure the sign-in resolver to bypass the user provisioning requirement in the {product-short} software catalog.
```

        sessionDuration: { hours: 24 }
----

`signIn` ::

`resolvers`:::

> After successful authentication, the user signing in must be resolved to an existing user in the {product-short} catalog. To best match users securely for your use case, consider configuring a specific resolver. Enter the resolver list to override the default resolver: `emailLocalPartMatchingUserEntityName`.

Reviewer 2: ```suggestion
After successfully authenticating, you must ensure the signing-in user is matched to an existing user in the {product-short} catalog. To securely match users for your specific use case, consider configuring a custom resolver. You can enter the resolver list to override the default resolver: `emailLocalPartMatchingUserEntityName`.
```

+
The authentication provider tries each sign-in resolver in order until it succeeds, and fails if none succeed.
+
WARNING: In production mode, only configure one resolver to ensure users are securely matched. 

`resolver`::::
Enter the sign-in resolver name.

> Available values:

Author: ```suggestion
Available resolvers:
```

> Available resolvers:

Author: ```suggestion
Available resolvers:

```

* `userIdMatchingUserEntityAnnotation`
* `emailLocalPartMatchingUserEntityName`
* `emailMatchingUserEntityProfileEmail`

`dangerouslyAllowSignInWithoutUserInCatalog: true`::::
Configure the sign-in resolver to bypass the user provisioning requirement in the {product-short} software catalog.
+
WARNING: Use `dangerouslyAllowSignInWithoutUserInCatalog` to explore {product-short} features, but do not use it in production.
+

> .`app-config-rhdh.yaml` fragment with optional field to allow authenticating users absent from the software catalog

Author: ```suggestion
.`app-config-rhdh.yaml` fragment with optional field to allow signing in users absent from the software catalog
```

        sessionDuration: { hours: 24 }
----

`signIn` ::

`resolvers`:::
After successful authentication, the user signing in must be resolved to an existing user in the {product-short} catalog. To best match users securely for your use case, consider configuring a specific resolver. Enter the resolver list to override the default resolver: `emailLocalPartMatchingUserEntityName`.
+
The authentication provider tries each sign-in resolver in order until it succeeds, and fails if none succeed.
+
WARNING: In production mode, only configure one resolver to ensure users are securely matched. 

`resolver`::::
Enter the sign-in resolver name.
Available resolvers:

* `userIdMatchingUserEntityAnnotation`
* `emailLocalPartMatchingUserEntityName`
* `emailMatchingUserEntityProfileEmail`

`dangerouslyAllowSignInWithoutUserInCatalog: true`::::

> +
> Configure the sign-in resolver to bypass the user provisioning requirement in the {product-short} software catalog.

Author: ```suggestion
Configure the sign-in resolver to bypass the user provisioning requirement in the {product-short} software catalog.
```
